src/comparsion.py

```diff
@@ -0,0 +1,159 @@
+import pandas as pd
+pd.set_option('display.max_columns', None)
+pd.set_option('display.width', 1000)
+
+import json
+import re
+from pathlib import Path
+
+# orginally clojure_mapping
+combined_mapping = {
+    'pre-increment': 'preIncr',
+    'operator-precedence': 'operator_precedence',
+    'conditional': 'conditional',
+    'preprocessor-in-statement': 'preprocessor_in_statement',
+    'logic-as-control-flow': 'logic_as_control_flow',
+    'type-conversion': 'type_conversion',
+    'comma-operator': 'comma_atoms',
+    'implicit-predicate': 'implicit_predicate',
+    'post-increment': 'postIncr',
+    'assignment-as-value': 'assignment_as_value',
+    'repurposed-variable': 'repurposed_variable',
+    'omitted-curly-braces' : 'omitted_curly_braces',
+    'macro-operator-precedence' : 'macro_operator_precedence',
+    'literal-encoding' : 'literal_encoding'
+}
+
+CLJnCQL = ["CLJ", "CQL"]
+CLJnCNL = ["CLJ", "CNL"]
+CQLnCNL = ["CQL", "CNL"]
+all = ["CLJ", "CQL", "CNL"]
+
+def count_plus_specific_columns(row, columns):
+    return (row[columns] == '+').sum()
+
+def compare_and_output(input_directory, output_directory, codeql_system_path, clojure_system_path, coccinelle_system_path, whichtwo):
+    codeQL_directory = Path(input_directory, "codeQL_data")
+    clojure_directory = Path(input_directory, "clojure_data/atomsClojure.csv")
+    coccinelle_directory = Path(input_directory, "coccinelle_data")
+    dataframes = {}
+    which_two = []
+    if whichtwo == "CQLnCNL":
+        which_two = CQLnCNL
+    elif whichtwo == "CLJvCQL":
+        which_two = CLJnCQL
+    elif whichtwo == "CLJvCNL":
+        which_two = CLJnCNL
+    else:
+        which_two = all
+
+    # Load CodeQL .sarif files and store them in dataframes dictionary
+
+    for file_path in Path(codeQL_directory).glob('*.sarif'):
+        file_content = file_path.read_text()
+        filename = Path(file_path).stem
+        dataframe_key = filename
+        json_value = json.loads(file_content)
+        p = pd.DataFrame(columns = ["Type", "File", "Line", "CQL", "CQL_Column", "CQL_CodeQL"])
+        rows = []
+        # These are the index paths of the data we are looking for. Each is located inside nested dictionaries.
+        for i in range(len(json_value["runs"][0]["results"])):
+            s = pd.Series({
+                "Type": filename,
```

Comment:
I deleted the [:-6] from the original code. I dont understand why but the atom name seems be missing six characters. After this change, the issue is solved but the cause unknown